### PR TITLE
Add tests and update project

### PR DESCRIPTION
**Pull request by countneuroman**
- Update project to NET 7.0
- add tests for success slove and incorrect size input data
- small refactoring code style and naming

### CODE REVIEW

File: SudokuSolverLib/Solver.cs [hunk 1]
```diff
@@ -47,19 +47,20 @@ public class Solver
...
         if (HorizontalPossibility(sudoku, row, value) && VerticalPossibility(sudoku, column, value) && SigmentPossibility(sudoku, row, column, value))
             return true;
         return false;
     }
-    bool HorizontalPossibility(int[,] sudoku, int row, int value)
+    private bool HorizontalPossibility(int[,] sudoku, int row, int value)
     {
-        bool result = true;
-        for (int i = 0; i < 9; i++)
+        var result = true;
```
Comment (Kartoshkin-btw) [inline SudokuSolverLib/Solver.cs]: bool is clearer

File: SudokuSolverLib/Solver.cs [hunk 2]
```diff
@@ -70,10 +71,10 @@ bool HorizontalPossibility(int[,] sudoku, int row, int value)
         }
         return result;
     }
-    bool VerticalPossibility(int[,] sudoku, int column, int value)
+    private bool VerticalPossibility(int[,] sudoku, int column, int value)
     {
-        bool result = true;
-        for (int i = 0; i < 9; i++)
+        var result = true;
```
Comment (Kartoshkin-btw) [inline SudokuSolverLib/Solver.cs]: bool is clearer

File: SudokuSolverLib/Solver.cs [hunk 3]
```diff
@@ -84,15 +85,15 @@ bool VerticalPossibility(int[,] sudoku, int column, int value)
         }
         return result;
     }
-    bool SigmentPossibility(int[,] sudoku, int row, int column, int value)
+    private bool SigmentPossibility(int[,] sudoku, int row, int column, int value)
     {
         int[,] sigment = Segments.GetSigment(row, column);
-        bool result = true;
-        for (int i = 0; i < 3; i++)
+        var result = true;
```
Comment (Kartoshkin-btw) [inline SudokuSolverLib/Solver.cs]: bool is clearer

File: SudokuSolverLib/Solver.cs (hunk 1 above)
Comment (countneuroman) [reply SudokuSolverLib/Solver.cs]: Fixed

File: SudokuSolverLib/Solver.cs (hunk 2 above)
Comment (countneuroman) [reply SudokuSolverLib/Solver.cs]: Fixed

File: SudokuSolverLib/Solver.cs (hunk 3 above)
Comment (countneuroman) [reply SudokuSolverLib/Solver.cs]: Fixed

**Pull request merged**